src/aux-records/RecordsServer.ts

```diff
@@ -2090,6 +2140,178 @@ export class RecordsServer {
+                                        return (
+                                            contract.accountCurrency ??
+                                            undefined === undefined
+                                        );
+                                    }
+                                },
+                                {
+                                    message:
+                                        'Contract must contain accountCurrency (only when status is "open").',
+                                    path: ['accountCurrency'],
+                                }
+                            ),
+                    })
+                )
+                .handler(async ({ contract }, context) => {
+                    const authUser = await this._validateSessionKey(
+                        context.sessionKey
+                    );
+                    if (!authUser.success) {
+                        return authUser;
+                    }
+                    // const result = await this._xpController.createContract({
+                    //     description: contract.description ?? null,
+                    //     accountCurrency: contract.accountCurrency,
+                    //     rate: contract.gigRate,
+                    //     offeredWorth: contract.gigs
+                    //         ? (contract.gigRate ?? 0) * contract.gigs
+                    //         : 0,
+                    //     status: contract.status,
+                    //     issuerUserId: { userId: authUser.userId },
+                    //     holdingUserId: contract.contractedUserId,
+                    //     creationRequestReceivedAt: Date.now(),
+                    // });
+                    // return result;
+                    return {
+                        success: true,
+                        message: 'Not implemented',
+                    };
+                }),
+
+            updateXpContract: procedure()
+                .origins('api')
+                .http('PUT', '/api/v2/xp/contract')
+                .inputs(
+                    z.object({
+                        contractId: z.string().min(1),
+                        newStatus: z.union([
+                            z.literal('open'),
+                            z.literal('draft'),
+                            z.literal('closed'),
+                        ]),
+                        //* Note this is the change of the description which can only be done when the contract is in draft status
+                        newDescription: z.string().optional(),
+                        receivingUserId: GetXpUserById,
+                    })
+                )
+                .handler(
+                    async (
+                        { contractId, newStatus, newDescription },
+                        context
+                    ) => {
+                        const authUser = await this._validateSessionKey(
```

Comment:
@LandonSiler This procedure should check to see if the xpController is supported as well.